components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,657 @@
+
+        $shippable = new ShippableObject();
+
+        $shipmentItem = new ShipmentItem();
+        $shipmentItem->setShippable($shippable);
+
+        $shipment = new Shipment();
+        $shipment->addItem($shipmentItem);
+
+        $ruleChecker = new ItemCountRuleChecker();
+        $ruleChecker->isEligible($shipment, $rule->getConfiguration()); // returns false, because
+        // quantity of shipping item in shipment is smaller than count from rule's configuration
+
+.. note::
+    This model implements the :ref:`component_shipping_checker_rule-checker-interface`. |br|
+    For more detailed information go to `Sylius API ItemCountRuleChecker`_.
+
+.. _Sylius API ItemCountRuleChecker: http://api.sylius.org/Sylius/Component/Shipping/Checker/ItemCountRuleChecker.html
+
+ShippingMethodEligibilityChecker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This class checks if shipping method rules are capable of shipping given subject.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\ShippingCategory;
+        use Sylius\Component\Shipping\Model\Shipment;
+        use Sylius\Component\Shipping\Model\ShipmentItem;
+        use Sylius\Component\Shipping\Model\ShippingMethodTranslation;
+        use Sylius\Component\Shipping\Model\RuleInterface;
+        use Sylius\Component\Shipping\Checker\Registry\RuleCheckerRegistry;
+        use Sylius\Component\Shipping\Checker\ItemCountRuleChecker;
+        use Sylius\Component\Shipping\Checker\ShippingMethodEligibilityChecker;
+
+        $rule = new Rule();
+        $rule->setConfiguration(array('count' => 0, 'equal' => true));
+        $rule->setType(RuleInterface::TYPE_ITEM_COUNT);
+
+
+        $shippingCategory = new ShippingCategory();
+        $shippingCategory->setName('Regular');
+
+        $hippingMethodTranslate = new ShippingMethodTranslation();
+        $hippingMethodTranslate->setLocale('en');
+        $hippingMethodTranslate->setName('First method');
+
+        $shippingMethod = new ShippingMethod();
+        $shippingMethod->setCategory($shippingCategory);
+        $shippingMethod->setCurrentLocale('en');
+        $shippingMethod->setFallbackLocale('en');
+        $shippingMethod->addTranslation($hippingMethodTranslate);
+
+        $shippingMethod->addRule($rule);
+
+        $shippable = new ShippableObject();
+        $shippable->setShippingCategory($shippingCategory);
+
+        $shipmentItem = new ShipmentItem();
+        $shipmentItem->setShippable($shippable);
+
+        $shipment = new Shipment();
+        $shipment->addItem($shipmentItem);
+
+        $ruleCheckerRegistry = new RuleCheckerRegistry();
+        $ruleChecker = new ItemCountRuleChecker();
+        $ruleCheckerRegistry->registerChecker(RuleInterface::TYPE_ITEM_COUNT, $ruleChecker);
+        $shippingMethodEligibilityChecker = new shippingMethodEligibilityChecker(
```

Comment:
Why this weird indent? We have a soft line limit. I'd put it all in a single like and perhaps make the variables shorter. ($shippingMethodEligibilityChecker -> $methodEligibilityChecker)